introspection-engine/connectors/sql-introspection-connector/src/calculate_datamodel.rs

```diff
@@ -61,6 +122,125 @@ pub fn calculate_model(schema: &SqlSchema) -> SqlIntrospectionResult<Datamodel>
+
+                        let name = match arity {
+                            FieldArity::List => inflector.pluralize(&model.name).camel_case(), // pluralize
+                            FieldArity::Optional => model.name.clone().camel_case(),
+                            FieldArity::Required => model.name.clone().camel_case(),
+                        };
+
+                        let field = Field {
+                            name,
+                            arity,
+                            field_type,
+                            database_name: None,
+                            default_value: None,
+                            is_unique: false,
+                            id_info: None,
+                            scalar_list_strategy: None,
+                            documentation: None,
+                            is_generated: false,
+                            is_updated_at: false,
+                        };
+
+                        fields_to_be_added.push((other_model.name.clone(), field));
+                    }
+                }
+                _ => {}
+            }
+        }
+    }
+
+    // add many to many relation fields
+    for table in schema
+        .tables
+        .iter()
+        .filter(|table| is_many_to_many_relation_table(&table))
+    {
+        // use foreign keys to find tables that are linked to??
+
+        let first = table.foreign_keys.get(0);
+        let second = table.foreign_keys.get(1);
+
+        match (first, second) {
+            (Some(f), Some(s)) => {
+                fields_to_be_added.push((
+                    s.referenced_table.clone(),
+                    create_many_to_many_field(f, &table.name),
+                ));
+                fields_to_be_added.push((
+                    f.referenced_table.clone(),
+                    create_many_to_many_field(s, &table.name),
+                ));
+            }
+            (_, _) => (),
+        }
+    }
+
+
+    // add scalar lists fields
+    for table in schema.tables.iter().filter(|table| is_scalar_list_table(&table)) {
+        let name = table.name.split('_').nth(1).unwrap();
+        let model = table.name.split('_').nth(0).unwrap();
+
+        let field_type = calculate_field_type(&table.columns.iter().find(|c| c.name == "value").unwrap(), &table);
+
+        let field = Field {
+            name: name.to_string(),
+            arity: FieldArity::List,
+            field_type,
+            database_name: None,
+            default_value: None,
+            is_unique: false,
+            id_info: None,
+            scalar_list_strategy: Some(ScalarListStrategy::Relation),
+            documentation: None,
+            is_generated: false,
+            is_updated_at: false,
+        };
+
+        fields_to_be_added.push((model.to_owned(), field));
+    }
+
+    for (model, field) in fields_to_be_added {
+        let model = data_model.find_model_mut(&model).unwrap();
```

Comment:
We always have the model in data_model?

Reply:
Yes, the is_prisma_scalar_list_table function verifies before this is being called

Reply:
Yes, it should always be there. I changed this to a match with a debug! print anyways just in case.